src/ophyd_async/plan_stubs/_nd_attributes.py

```diff
@@ -0,0 +1,67 @@
+from typing import Sequence
+from xml.etree import cElementTree as ET
+
+import bluesky.plan_stubs as bps
+
+from ophyd_async.core._device import Device
+from ophyd_async.epics.adcore._core_io import NDArrayBaseIO
+from ophyd_async.epics.adcore._utils import (
+    NDAttributeDataType,
+    NDAttributeParam,
+    NDAttributePv,
+)
+
+
+def setup_ndattributes(
+    device: NDArrayBaseIO, ndattributes: Sequence[NDAttributePv | NDAttributeParam]
+):
+    xml_text = yield from bps.rd(device.nd_attributes_file)
+    if xml_text == "":
+        xml_text = ET.Element("Attributes")
```

Comment:
I'm not sure why you made this change to check what the existing nd_attributes_file was?